src/api/tokenList/TokenListApi.ts

```diff
@@ -13,21 +26,67 @@ export interface TokenList {
 export class TokenListApiImpl implements TokenList {
   public networkIds: number[]
   private _tokensByNetwork: { [networkId: number]: TokenDetails[] }
+  private _tokenAddressNetworkSet: Set<string>
 
   public constructor(networkIds: number[]) {
     this.networkIds = networkIds
 
     // Init the tokens by network
     this._tokensByNetwork = {}
+    this._tokenAddressNetworkSet = new Set<string>()
+
     networkIds.forEach(networkId => {
       // initial value
-      this._tokensByNetwork[networkId] = getTokensByNetwork(networkId)
+      const tokenList = getTokensByNetwork(networkId).concat(this.loadUserTokenList(networkId))
+      this._tokensByNetwork[networkId] = tokenList
+
+      tokenList.forEach(({ address }) => {
+        this._tokenAddressNetworkSet.add(
+          TokenListApiImpl.constructAddressNetworkKey({ tokenAddress: address, networkId }),
+        )
+      })
     })
   }
 
+  public hasToken(params: HasTokenParams): boolean {
+    return this._tokenAddressNetworkSet.has(TokenListApiImpl.constructAddressNetworkKey(params))
+  }
+
   public getTokens(networkId: number): TokenDetails[] {
     return this._tokensByNetwork[networkId] || []
   }
+
+  private static constructAddressNetworkKey({ tokenAddress, networkId }: HasTokenParams): string {
+    return tokenAddress.toLowerCase() + '|' + networkId
+  }
+
+  private static getUserTokenListName(networkId: number): string {
+    return 'USER_TOKEN_LIST_' + networkId
```

Comment:
nit: why the word `USER`? since this is for the local storage, it's implied. Why not just `TOKENS_` + networkId?

Reply:
but we will have our own `TOKEN_LIST` later when we start fetching ids and persisting in localStorage
User should have its own list